src/peer_connection.cpp

```diff
@@ -1319,8 +1318,11 @@ namespace libtorrent {
 				}
 				// find the lowest ranking peer and disconnect that
 				peer_connection* p = other_t->find_lowest_ranking_peer();
-				p->disconnect(errors::too_many_connections, operation_t::bittorrent);
-				peer_disconnected_other();
+				if (p != nullptr)
```

Comment:
@arvidn do I need to put an `else` here? or is this correct?

Reply:
one of the connections need to be disconnected, so there needs to be an else block doing:
```
 disconnect(errors::too_many_connections, operation_t::bittorrent);
 return;
```

I believe

Reply:
Ok, I will add that